tests/e2e/gc_rwx_volume_health.go

```diff
@@ -0,0 +1,339 @@
+			ginkgo.By("poll for health status annotation")
+			err = pvcHealthAnnotationWatcher(ctx, client, pvclaim, healthStatusAccessible)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		//Add a check to validate CnsVolumeMetadata crd
+		err = waitAndVerifyCnsVolumeMetadata4GCVol(fcdIDInCNS, pvcNameInSV, pvclaim, pv, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	/*
+	   Annotations should be accessble to all the RWX PVCs which are in bound state.
+	   1. Create a PVC using invalid SC
+	   2. PVC should be in pending state
+	   3. Verify health annotation is not added on the PVC
+	   4. Delete PVC
+	*/
+	ginkgo.It("Verify volume health annotation is not added for PVC in pending", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = nonVsanStoragePolicyName
+
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to fail as the storage policy mentioned in Storage class " +
+			"has non-vSAN compliant datastores")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+			pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute/2)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Sleeping for %v minutes to allow volume health check to be triggered",
+			healthStatusWaitTime))
+		time.Sleep(healthStatusWaitTime)
+
+		ginkgo.By("Expect health status to be not added")
+		pvc, err := client.CoreV1().PersistentVolumeClaims(pvclaim.Namespace).Get(ctx,
+			pvclaim.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvc.Annotations[volumeHealthAnnotation]).Should(gomega.BeEmpty())
+	})
+
+	/*
+	   Verify RWX volume health is accessible on vsan health service restart
+	   1. Create SC
+	   2. Create a PVC with "ReadWriteMany" access mode using the SC from above in GC
+	   3. Wait for PVC to be Bound in GC
+	   4. Verify if the mapping PVC is also bound in the SV cluster using the volume handler
+	   5. Verify CnsVolumeMetadata CRD is created
+	   6. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+	   7. Bring vsan-health down
+	   8. Wait for the health status interval time
+	   9. Verify no health annotation is added on the PVC
```

Comment:
what about the case where the health='accessible' and then cns goes down
In the subsequent health-status interval, the health should turn 'inaccessible'

Reply:
cns down usecases are not part of this PR